src/pixelator/plot/__init__.py

```diff
@@ -44,6 +54,202 @@ def _calculate_densities(coordinates, distance_cutoff, unit_sphere_surface):
+
+    assert all(
+        [
+            data["umi_per_upia"].dtype == "float64",
+            data["tau"].dtype == "float64",
+            data["tau_type"].dtype in ["object", "category"],
+        ]
+    ), "Invalid data types"
+
+    if group_by is not None:
+        if group_by not in data.columns:
+            raise ValueError(f"'{group_by}' is missing")
+        assert data[group_by].dtype in [
+            "object",
+            "category",
+        ], "'group_by' must be a character or factor"
+
+    # Define palette
+    palette = {"high": ORANGERED2, "low": SKYBLUE3, "normal": LIGHTGREY}
+
+    # create plot
+    grid = (
+        sns.FacetGrid(
+            data,
+            col=group_by,
+            hue="tau_type",
+            palette=palette,
+            height=4,
+        )
+        if group_by is not None
+        else sns.FacetGrid(data, hue="tau_type", palette=palette, height=4)
+    )
+    grid.map(sns.scatterplot, "tau", "umi_per_upia")
+    grid.set(yscale="log")
+    grid.set_axis_labels("Marker specificity (Tau)", "Pixel content (UMI/UPIA)")
+    grid.add_legend(title="Tau Type")
+
+    return plt.gcf(), plt.gca()
+
+
+def plot_2d_graph(
+    component_graph: Graph,
+    marker: str = "pixel_type",
+    layout_algorithm: Literal[
+        "fruchterman_reingold", "kamada_kawai", "pmds"
+    ] = "fruchterman_reingold",
+    show_edges=False,
+    log_scale=True,
+    node_size=10.0,
+    edge_width=1.0,
+    show_Bnodes=False,
+    cmap="cool",
+    cache_layout: bool = False,
+) -> Tuple[plt.Figure, plt.Axes]:
+    """Plot a 2D graph representation of a component graph.
+
+    Args:
+    ----
+        component_graph (Graph): The component graph to be plotted.
+        marker (str, optional): The marker used to color the nodes. Defaults to pixel_type.
+        layout_algorithm (Literal["fruchterman_reingold", "kamada_kawai", "pmds"], optional):
+            The layout algorithm used to position the nodes. Defaults to "fruchterman_reingold".
+        show_edges (bool, optional): Whether to show the edges in the plot. Defaults to False.
+        log_scale (bool, optional): Whether to use a logarithmic scale for the marker values. Defaults to True.
+        node_size (float, optional): The size of the nodes. Defaults to 10.0.
+        edge_width (float, optional): The width of the edges. Defaults to 1.0.
+        show_Bnodes (bool, optional): Whether to show the B-nodes in the plot. Defaults to False.
+        cmap (str, optional): The colormap used for coloring the nodes. Defaults to "cool".
+        cache_layout (bool, optional): Whether to cache the layout coordinates. Defaults to False.
+
+    Returns:
+    -------
+        Tuple[plt.Figure, plt.Axes]: The matplotlib figure and axes objects.
+
+    """
+    coordinates = component_graph.layout_coordinates(
+        layout_algorithm=layout_algorithm,
+        cache=cache_layout,
+        only_keep_a_pixels=not show_Bnodes,
+    )
+    filtered_coordinates = coordinates
```

Comment:
MAybe we can do a live review from this line. I will send in my review so you can see my other comments.